### ISSUE TITLE
[BUG] in integration tests with VW

### ISSUE THREAD

**Issue by miguelgfierro** [Microsoft/Recommenders#716]
### Description

```
tests/integration/test_criteo.py .                                       [  6%]
tests/integration/test_movielens.py .........                            [ 66%]
tests/integration/test_notebooks_python.py ....F                         [100%]

=================================== FAILURES ===================================
______________ test_vw_deep_dive_integration[1m-expected_values0] ______________

notebooks = {'als_deep_dive': '/data/home/recocat/cicd/7/s/notebooks/02_model/als_deep_dive.ipynb', 'als_pyspark': '/data/home/rec...baseline_deep_dive.ipynb', 'data_split': '/data/home/recocat/cicd/7/s/notebooks/01_prepare_data/data_split.ipynb', ...}
size = '1m'
expected_values = {'exp_var': 0.266197, 'mae': 0.686706, 'map': 0.186626, 'ndcg': 0.681827, ...}

    @pytest.mark.integration
    @pytest.mark.parametrize(
        "size, expected_values",
        [
            (
                "1m",
                dict(
                    rmse=0.958504,
                    mae=0.686706,
                    rsquared=0.26606,
                    exp_var=0.266197,
                    map=0.186626,
                    ndcg=0.681827,
                    precision=0.5748,
                    recall=0.186626,
                ),
            )
        ],
    )
    def test_vw_deep_dive_integration(notebooks, size, expected_values):
        notebook_path = notebooks["vowpal_wabbit_deep_dive"]
        pm.execute_notebook(
            notebook_path,
            OUTPUT_NOTEBOOK,
            kernel_name=KERNEL_NAME,
            parameters=dict(MOVIELENS_DATA_SIZE=size, TOP_K=10),
        )
        results = pm.read_notebook(OUTPUT_NOTEBOOK).dataframe.set_index("name")["value"]
    
        for key, value in expected_values.items():
>           assert results[key] == pytest.approx(value, rel=TOL, abs=ABS_TOL)
E           assert 0.5077596367584887 == 0.186626 ± 5.0e-02
E            +  where 0.186626 ± 5.0e-02 = <function approx at 0x7f6355d952f0>(0.186626, rel=0.05, abs=0.05)
E            +    where <function approx at 0x7f6355d952f0> = pytest.approx

```
original error: https://msdata.visualstudio.com/AlgorithmsAndDataScience/_build/results?buildId=2781053

### In which platform does it happen?

Azure Data Science Virtual Machine

### Expected behavior (i.e. solution)

tests passing

**Comment by miguelgfierro** [comment]
I've been looking at this, the failure comes in the integration tests of "1m". I found that 19 days ago we changed the metrics, [originally we had](https://github.com/Microsoft/Recommenders/blob/0554b7ff853e49e109c0ba71144be3c4a8a1181e/tests/integration/test_notebooks_python.py#L121):
```
"1m",
            dict(
                rmse=0.9555,
                mae=0.68493,
                rsquared=0.26547,
                exp_var=0.26615,
                map=0.50635,
                ndcg=0.99966,
                precision=0.92684,
                recall=0.50635,
            ),
```
tbh the rating metrics seemed too high, and [now we have](https://github.com/Microsoft/Recommenders/blob/staging/tests/integration/test_notebooks_python.py#L121):

```
            "1m",
            dict(
                rmse=0.958504,
                mae=0.686706,
                rsquared=0.26606,
                exp_var=0.266197,
                map=0.186626,
                ndcg=0.681827,
                precision=0.5748,
                recall=0.186626,
            ),
```
If you run now the notebook, we will get the first set of metrics again, @gramhagen do you remember why we changed the metrics?

The only change I could find in the notebook is `pred_df.drop("rating", axis=1, inplace=True)` in cell 3. This is the [new notebook](https://github.com/Microsoft/Recommenders/blob/staging/notebooks/02_model/vowpal_wabbit_deep_dive.ipynb) and this is the [old one](https://github.com/Microsoft/Recommenders/blob/90b7bd1417efbd1455f8ed588bff56a26698637f/notebooks/02_model/vowpal_wabbit_deep_dive.ipynb)

**Comment by gramhagen** [comment]
this is a little fuzzy, i think it was due to the dataframes having identical column names ('rating') when we merged them inside the metrics code the true rating column was renamed, it looks like I'm not dropping that column in the final calculation used for the metrics?
perhaps the latest changes to the metrics code exposed this? I'll look into it, we may need to check for this case in the metrics calculations.

**Comment by gramhagen** [comment]
yeah, couple of problems. if a column sharing the rating column name exists in the prediction dataframe it generates errors when calculating metrics. I've corrected this issue in #713, but there are also problems with how I'm generating the all pair evaluation set. I need to update the test metrics for all the versions of the notebook.  they are much lower now =(  ... but as you mention they were inexplicable high before, so it seems better

**Comment by gramhagen** [comment]
fixed in #722
